### PR TITLE
Feature/api server token statistics

### PR DESCRIPTION
**Pull request by OBorce**
add endpoints for statistics for coins and tokens, returning
- ciruculating_amount the total available circulating amount
- preminted the total amount of coins that were preminted in the genesis block
- burned the total amount of coins/tokens that got burned or spent on issuance fees
- staked the total amount of coins currently being staked in pools or delegations

Added new endpoint GET token that returns a list of token ids with length and offset query parameters

### CODE REVIEW

File: api-server/api-server-common/src/storage/storage_api/mod.rs [hunk 1]
```diff
@@ -64,6 +68,47 @@ pub enum ApiServerStorageError {
     TimestampToHigh(BlockTimestamp),
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug)]
+pub enum CoinOrTokenStatistic {
+    TotalSupply,
+    Staked,
+    Burned,
+    Preminted,
```
Comment (TheQuantumPhysicist) [inline api-server/api-server-common/src/storage/storage_api/mod.rs:76]: You mean Unminted?

File: api-server/api-server-common/src/storage/storage_api/mod.rs [hunk 2]
```diff
@@ -64,6 +68,47 @@ pub enum ApiServerStorageError {
     TimestampToHigh(BlockTimestamp),
 }
 
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug)]
+pub enum CoinOrTokenStatistic {
+    TotalSupply,
```
Comment (azarovh) [inline api-server/api-server-common/src/storage/storage_api/mod.rs:73]: It seems like it should be `CirculatingSupply` to conform with chainstate terminology where TotalSupply is the limit for minting and Circulating is current amount minted/unminted. But I'm confused why this arm is used for coins in transfer outputs.

File: api-server/api-server-common/src/storage/storage_api/mod.rs (hunk 1 above)
Comment (OBorce) [reply api-server/api-server-common/src/storage/storage_api/mod.rs:76]: Preminted are the coins from genesis

File: api-server/api-server-common/src/storage/storage_api/mod.rs (hunk 2 above)
Comment (OBorce) [reply api-server/api-server-common/src/storage/storage_api/mod.rs:73]: the transfer output is probably the arm from block reward, those are from genesis.

**Pull request merged**